architecture.md

```diff
@@ -55,4 +55,19 @@ explicit configuration.
 ![trusted-source-compile-container](./assets/source-container-trusted.png)
 _Trusted Components are included in the same container context as their parent Component._
 
-### Component Callbacks (WIP)
+### Component Callbacks
+Callbacks between Components (e.g. functions passed via `props`) are supported in BOS Web Engine, though the exact behavior is
+dependent on whether the Components are within the same container. Callbacks between two Components _in the same container_
+work without constraints, but there are caveats for callbacks _between containers_ as the caller and callee exist in completely
+independent contexts. These inter-container callbacks must be invoked (and return data) using the message passing interface
```

Comment:
```suggestion
independent contexts. Under the hood, these inter-container callbacks must be invoked (and return data) using the message passing interface
```

nit, this phrasing makes it more clear this is something done by the engine and not something a user has to rig up